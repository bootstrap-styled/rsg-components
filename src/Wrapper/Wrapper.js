import { Component } from 'react';
import PropTypes from 'prop-types';

export default class Wrapper extends Component {
  static propTypes = {
    children: PropTypes.node.isRequired,
    onError: PropTypes.func.isRequired,
  };

  componentDidCatch(error) {
    this.props.onError(error);
  }

  render() {
    return this.props.children;
  }
}
