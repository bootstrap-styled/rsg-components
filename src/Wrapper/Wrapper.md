The wrapper component is used for each rendered example

```jsx harmony
import S from 'rsg-components';
<Wrapper>
  Wrapped{S.toString()}
</Wrapper>
```
