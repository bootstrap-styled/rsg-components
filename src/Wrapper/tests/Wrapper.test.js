import { shallow } from 'enzyme';
import React from 'react';
import Wrapper from '../Wrapper';

it('should render children', () => {
  const children = <span>Hello</span>;
  const actual = shallow(<Wrapper onError={() => {}}>{children}</Wrapper>);

  expect(actual).toMatchSnapshot();
});

it('should call onError handler when React invokes error handler', () => {
  const onError = jest.fn();
  const actual = shallow(<Wrapper onError={onError}>blah</Wrapper>);

  // faux error
  actual.instance().componentDidCatch('err');

  expect(onError).toHaveBeenCalledTimes(1);
  expect(onError).toHaveBeenCalledWith('err');
});
