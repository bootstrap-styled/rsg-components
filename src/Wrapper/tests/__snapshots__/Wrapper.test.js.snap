// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render children 1`] = `ShallowWrapper {}`;
